### DISCUSSION TITLE
Cloudwatch: Insights Report query support

### DISCUSSION THREAD

**Discussion by mourya92** [grafana/grafana#60318] [Data Sources]
Hi Grafana team, 

**Issue**
I am trying to query [Contributor Insights Report](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetInsightRuleReport.html) using Cloudwatch datasource plugin to get Top Talkers.  But looks like the plugin does not support querying the Insights Report though it’s [public documentation](https://grafana.com/docs/grafana/latest/datasources/aws-cloudwatch/) says to add that permission("cloudwatch:GetInsightRuleReport"). 

There is a secondary use for this permission which is - to query Contributor Insights Report **Stats** using [Metric Math Expression](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContributorInsights-GraphReportData.html) (INSIGHT_RULE_METRIC) but not the report itself that includes actual contributor list.

**Example**
If customer setup a Contributor Insights Rule to find Top N UserAgents thats receiving 500 errors from service, Cloudwatch offers GetInsightRuleReport API that can be used to extract these top-talkers and datapoints as Time-series data. Today Cloudwatch plugin supports `INSIGHT_RULE_METRIC` Metric Query that can be used to get stats of the contributors list (like number of unique contributors, sum of the values from all contributors etc.,) but not the contributors list itself because this is a different API. 

**Ask**
Add support to view graphs of report data and list of contributors found by Cloudwatch Contributor Insight rules using `GetInsightRuleReport` API as the plugin already has IAM permissions to get the report but missing support to that API.

Note: We cannot query insights report data using Cloudwatch GetMetricData API. GetMetricData API can only be used to get stats of Insights report data but not the actual report itself.

---

_Somewhat related discussion_ : https://github.com/grafana/grafana/discussions/39190

**Comment by sarahzinger** [comment]
Hi @mourya92 as part of our work to move from [discussions to issues](https://github.com/grafana/grafana/discussions/73424), I've opened up [an issue](https://github.com/grafana/grafana/issues/77952) for you and put it in our backlog to consider for future work. Thanks for the suggestion! If you have any other thoughts or ideas feel free to add them to that issue!
